DEVELOPMENT.md

```diff
@@ -59,3 +59,47 @@ To build the image, which requires access to Red Hat CI registry, run:
 ```shell
 make docker-build
 ```
+
+# E2E Testing
+
+> [!NOTE]
+> For macOS users, you will need to install `gsed` using `brew install gnu-sed`
+
+## Running in KinD
+
+Firstly, bring up the environment:
+
+```shell
+make mco-kind-env
+```
+
+This will create a cluster with MCO installed and the operator running.
+
+Then, run the tests:
+
+```shell
+make e2e-tests-in-kind
+```
+
+By default, if the tests fail, the ACM components will be removed.
+We can bypass this behaviour by running:
+
+```shell
+SKIP_UNINSTALL_STEP=true make e2e-tests-in-kind
+```
+
+Likewise, we can also bypass the environment setup by running:
+
+```shell
+SKIP_INSTALL_STEP=true make e2e-tests-in-kind
+```
+
+After running the tests, there will be some changes made to your environment.
+We can clean up the environment by running:
```

Comment:
This is not ideal. It will be better to copy all the required manifests into a temp directory that we can exclude and act on the copies. 

Unfortunately these paths are deeply embedded within the tests so I will create a follow-up in order to not bloat this further than it already is!